ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineGetBlobsV3.java

```diff
@@ -0,0 +1,141 @@
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.Counter;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EngineGetBlobsV3 extends ExecutionEngineJsonRpcMethod {
+  private static final Logger LOG = LoggerFactory.getLogger(EngineGetBlobsV3.class);
+  public static final int REQUEST_MAX_VERSIONED_HASHES = 128;
+
+  private final TransactionPool transactionPool;
+  private final Counter requestedCounter;
+  private final Counter availableCounter;
+  private final Counter hitCounter;
+  private final Counter missCounter;
+
+  public EngineGetBlobsV3(
+      final Vertx vertx,
+      final ProtocolContext protocolContext,
+      final EngineCallListener engineCallListener,
+      final TransactionPool transactionPool,
+      final MetricsSystem metricsSystem) {
+    super(vertx, protocolContext, engineCallListener);
+    this.transactionPool = transactionPool;
+    // create counters
+    this.requestedCounter =
+        metricsSystem.createCounter(
+            BesuMetricCategory.RPC,
+            "execution_engine_getblobs_requested_total",
+            "Number of blobs requested via engine_getBlobsv3");
+    this.availableCounter =
+        metricsSystem.createCounter(
+            BesuMetricCategory.RPC,
+            "execution_engine_getblobs_available_total",
+            "Number of blobs requested via engine_getBlobsv3 that are present in the blob pool");
+    this.hitCounter =
+        metricsSystem.createCounter(
+            BesuMetricCategory.RPC,
+            "execution_engine_getblobs_hit_total",
+            "Number of calls to engine_getBlobsv3 that returned at least one blob");
+    this.missCounter =
+        metricsSystem.createCounter(
+            BesuMetricCategory.RPC,
+            "execution_engine_getblobs_miss_total",
+            "Number of calls to engine_getBlobsv3 that returned zero blobs");
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ENGINE_GET_BLOBS_V3.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext) {
+    final VersionedHash[] versionedHashes = extractVersionedHashes(requestContext);
+    if (versionedHashes.length > REQUEST_MAX_VERSIONED_HASHES) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(),
+          RpcErrorType.INVALID_ENGINE_GET_BLOBS_TOO_LARGE_REQUEST);
+    }
+    requestedCounter.inc(versionedHashes.length);
+    final List<BlobAndProofV2> result =
+        Stream.of(versionedHashes).map(this::getBlobAndProofOrNull).toList();
+    long available = result.stream().filter(java.util.Objects::nonNull).count();
+    availableCounter.inc(available);
+    if (available > 0) {
+      hitCounter.inc();
+    } else {
+      missCounter.inc();
+    }
```

Comment:
This counter may be most useful if we increment by `versionedHashes.length - available`. At least for us, the p2p networking team! Tracking full miss responses may be useful too in a separate counter.